apps/system/utils/utils_kill.c

```diff
@@ -235,24 +238,62 @@ int utils_kill(int argc, char **args)
 #endif
 
 #if defined(CONFIG_ENABLE_KILLALL)
-static void utils_killall_handler(FAR struct tcb_s *tcb, FAR void *arg)
+static void killall_send_signal(char *buf, void *arg)
 {
+	int i;
+	stat_data stat_info[PROC_STAT_MAX];
+	char *name;
+	int pid;
 	struct utils_killall_arg_s *killall_arg = (struct utils_killall_arg_s *)arg;
+	if (killall_arg == NULL) {
+		printf("Failed to send signal, because of invalid parameter.\n");
+		return;
+	}
+
+	stat_info[0] = buf;
+	for (i = 0; i < PROC_STAT_MAX - 1; i++) {
+		stat_info[i] = strtok_r(stat_info[i], " ", &stat_info[i + 1]);
+	}
+	name = stat_info[PROC_STAT_NAME];
+	pid = atoi(stat_info[PROC_STAT_PID]);
 
-	if (strncmp(tcb->name, killall_arg->name, CONFIG_TASK_NAME_SIZE) == 0) {
+	if (strncmp(name, killall_arg->name, CONFIG_TASK_NAME_SIZE) == 0) {
 		killall_arg->count++;
-		if (send_signal(tcb->pid, killall_arg->signo) != OK) {
+		if (send_signal(pid, killall_arg->signo) != OK) {
 			*(killall_arg->ret) = ERROR;
 		}
 	}
 }
 
+static int killall_read_proc(FAR struct dirent *entryp, FAR void *arg)
+{
+	char *filepath;
+	char buf[KILLALL_BUFLEN];
+	int ret;
+
+	asprintf(&filepath, "%s/%s/%s", PROCFS_MOUNT_POINT, entryp->d_name, "stat");
+	ret = utils_readfile(filepath, buf, KILLALL_BUFLEN, arg, killall_send_signal);
+	if (ret < 0) {
+		printf("Failed to read %s\n", filepath);
+		free(filepath);
+		return ERROR;
+	}
+
+	free(filepath);
```

Comment:
Extra empty lines needed both before and after this one?

Reply:
Removed. Thanks.